code/parachain/frame/composable-maths/src/dex/constant_product.rs

```diff
@@ -123,6 +123,77 @@ where
 	Ok(ai)
 }
 
+/// Compute the amount of the input token given the amount of the output token.
+///
+/// If `Ok`, returns a tuple containing `(a_sent, fee)`.
+/// To get `a_sent` without accounting for the fee, set `f = 0`.
+/// Amount in, round up results.
+///
+/// **NOTE:** Weights must already be normalized.
```

Comment:
what's meant by normalized here?

Reply:
sum of weights = 1

Reply:
We can obviously clarify that in the comment for lay readers not familiar with dex stuff.

Reply:
If the sum of weights is 1, then we only need one input parameter and can use [`PerThing::left_from_one()`](https://paritytech.github.io/substrate/master/sp_runtime/trait.PerThing.html#method.left_from_one) to get `w_i` and `w_o`

Reply:
Eventually, we'll have more than two tokens to a pool @benluelo. They will still be normalized, but to the scope of all tokens in the pool

Reply:
ah, so normalized between 0 and 1. makes sense